lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -778,6 +779,126 @@ class DecomposeAtenConv2dOp : public OpRewritePattern<AtenConv2dOp> {
+  }
+};
+} // namespace
+
+// Decompose aten.convolution_backward_overrideable to aten.convolution_backward
+namespace {
+class DecomposeAtenConvolutionBackwardOp
+    : public OpRewritePattern<AtenConvolutionBackwardOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenConvolutionBackwardOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    MLIRContext *context = op.getContext();
+    Value ci0 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
+    Value ci1 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    Value cNone = rewriter.create<Torch::ConstantNoneOp>(loc);
+    Value cFalse = rewriter.create<Torch::ConstantBoolOp>(
+        loc, rewriter.getBoolAttr(false));
+
+    Value gradOutput = op.grad_output();
+    Value input = op.input();
+    Value weight = op.weight();
+    auto gradRank = getTensorRank(gradOutput);
+    SmallVector<Value> padding;
+    getListConstructElements(op.padding(), padding);
+    SmallVector<int64_t> paddingInts;
+    if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only constant padding is supported");
+
+    if (gradRank != 4)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only 2D convolutions supported");
+
+    // Rotate weight
+    SmallVector<Value> axis;
+    for (auto i = 2; i < gradRank; i++)
+      axis.push_back(rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(i)));
+    Value axisList = rewriter.create<Torch::PrimListConstructOp>(
+        loc, Torch::ListType::get(Torch::IntType::get(op.getContext())), axis);
+    weight = rewriter.create<Torch::AtenFlipOp>(loc, weight.getType(), weight,
+                                                axisList);
+
+    // Transpose inputs
+    Value gradOutputTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, gradOutput.getType(), gradOutput, ci0, ci1);
+    Value inputTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, input.getType(), input, ci0, ci1);
+    Value weightTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, weight.getType(), weight, ci0, ci1);
+
+    // Convolve grad_output with weight (padding = [1, 1])
+    Value gradInputPadding = rewriter.create<Torch::PrimListConstructOp>(
+        loc, ListType::get(IntType::get(context)),
+        SmallVector<Value>(gradRank - 2, ci1));
+    Value gradInput = rewriter.create<Torch::AtenConvolutionOp>(
+        loc, op.getResultTypes()[0], gradOutput, weightTransposed, cNone,
+        op.stride(), gradInputPadding, op.dilation(), op.transposed(),
+        op.output_padding(), op.groups());
+
+    // Shrink input_grad
+    for (auto i = 2; i < gradRank; i++) {
+      if (paddingInts[i - 2] != 0) {
```

Comment:
nit: this can be easily checked at runtime with cmp and select. Then we can remove the restriction for int constant.

Reply:
I'm familiar with cmp and select in arith, but not in Torch. What ops do we use for that?